File: Examples/python/json_output.py
import sys
import os

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..")))

from Client.llmClient import ask_llm, build_strict_json_prompt
import json

def run_task():
    task = "Extract product details from the description."
    context = "The new UltraTab G15 features 128GB storage and a 12-inch OLED screen for $499."
    
    # Define the structure you WANT
    schema = {
        "product_name": "string",
        "storage_gb": "number",
        "screen_size_inches": "number",
        "price_usd": "number"
    }

    # 1. Build the prompt
    prompt = build_strict_json_prompt(task, context, schema)

    # 2. Call the bridge
    response = ask_llm(prompt)

    if response.get('ok'):
        try:
            # Parse the result string
            data = json.loads(response['data'])
            print("Data Extracted Successfully:")
            print(json.dumps(data, indent=4))
        except Exception as e:
            print(f"JSON Parse Error: {e}")
            print("Raw Response:", response.get('data'))

if __name__ == "__main__":
    run_task()

File: Examples/javascript/json_output.js
import { askLLM, buildStrictJSONPrompt } from '../../Client/llmClient.js';

async function performAITask() {
  const task = "Extract contact information from the text.";
  const context = "Reach out to John Doe at john.doe@example.com or call 555-0199.";
  
  // Define the structure you WANT
  const schema = {
    name: "string",
    email: "string",
    phone: "string",
    confidence_score: "number (0-1)"
  };

  const prompt = buildStrictJSONPrompt({ task, context, schema });
  const response = await askLLM(prompt);

  if (response.ok) {
    try {
      const data = JSON.parse(response.data);
      console.log("Extracted Data:", data);
    } catch (e) {
      console.error("Parsing Error. Raw Text:", response.data);
    }
  }
}

await performAITask()

